rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -331,6 +367,157 @@ def calculate_claimed_airdrops(
         ).fetchall()
 
 
+def process_airdrop_with_api_data(
+        addresses: Sequence[ChecksumEvmAddress],
+        airdrop_data: AirdropAPIData,
+        protocol_name: str,
+        airdrops_without_decoder: set[str],
+        tolerance_for_amount_check: FVal,
+) -> tuple[list[AirdropClaimEventQueryParams], dict[ChecksumEvmAddress, dict]]:
+    """
+    Returns information for airdrops that can be queried only using remote APIs.
+    """
+    timeout_tuple = CachedSettings().get_timeout_tuple()
+    found_data: dict[ChecksumEvmAddress, dict] = defaultdict(lambda: defaultdict(dict))
+    temp_airdrop_tuples = []
+    for address in addresses:
+        try:
+            response = requests.get(
+                url=airdrop_data.api_url.format(address=address),
+                timeout=timeout_tuple,
+            )
+            data: dict[str, Any] = response.json()
+        except requests.exceptions.RequestException as e:
+            log.error(f'Failed to query {protocol_name} airdrop for address {address} due to {e}')
+            continue
+
+        if data.get('status') != 'Complete':
```

Comment:
the `status` key in the response is protcol specific right? while means we can't use that for all the protocols with some API. Maybe check status_code or pass this arbitrary value/path in the index?